lib/zzhc/zzhc_port.c

```diff
@@ -0,0 +1,364 @@
+#include <at_cmd.h>
+#include <at_notif.h>
+#include <base64.h>
+#include <secure_services.h>
+#include <cJSON.h>
+#include <cJSON_os.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <settings/settings.h>
+
+#define AT_PARAMS_MAX     10        /** Max. # of AT-params to parse */
+#define ICCID_FILE_ID     0x2B48    /** File ID of stored ICCID */
+#define THREAD_PRIORITY   K_PRIO_PREEMPT(CONFIG_ZZHC_THREAD_PRIO)
+#define STACK_SIZE        CONFIG_ZZHC_STACK_SIZE
+#define FICR_REV_OFFSET   0x134
+
+/** Internal thread stack. */
+K_THREAD_STACK_DEFINE(thread_stack, STACK_SIZE);
+
+/** Cached storage of ICCID. */
+static char ueiccid[ICCID_LEN];
+
+/** Name of settings */
+static char const settings_name[] = "zzhc";
+
+#ifndef LOG_DBG
+#define LOG_DBG(...) \
+{\
+    printf(__VA_ARGS__);\
+    printf("\n");\
+}
+#endif
+
+/**@brief POSIX sem_init. */
+static int sem_init(void *sem, int pshared, unsigned int value)
+{
+	ARG_UNUSED(pshared);
+
+	if (sem == NULL) {
+		return -EINVAL;
+	}
+
+	k_sem_init(sem, value, 1);
+	return 0;
+}
+
+/**@brief POSIX sem_wait() */
+static int sem_wait(void *sem)
+{
+	return k_sem_take(sem, K_FOREVER);
+}
+
+/**@brief POSIX sem_post() */
+static int sem_post(void *sem)
+{
+	k_sem_give(sem);
+	return 0;
+}
+
+/**@brief Sleep (seconds). */
+static void sleep(int t)
+{
+	(void)k_sleep(t * 1000);
+}
+
+/**@brief Malloc. */
+static void *this_malloc(size_t size)
+{
+	return k_malloc(size);
+}
+
+/**@brief Free. */
+static void this_free(void *ptr)
+{
+	k_free(ptr);
+}
```

Comment:
I don't see the value in these wrappers. k_free should be invoked directly instead of indirectly.

Reply:
The same applies for other wrappers here.

Reply:
zzhc.c and zzhc_port.c will be merged as one file right before the end of this review, in this PR.
Then you will not see any indirect calls like this.
In the next few days they still need to be kept. Sorry about this.